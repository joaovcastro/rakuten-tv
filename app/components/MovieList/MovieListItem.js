import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faStar, faUser } from '@fortawesome/free-solid-svg-icons';

const Icon = styled(FontAwesomeIcon)`
  margin-right: 7px;
`;

const MovieListItem = elem => {
  const { item, onClick } = elem;
  return (
    <Link to={`/movie/${item.id}`}>
      <div className="grid" onClick={onClick}>
        <img
          className="movie-poster"
          src={item.images.artwork}
          width="225px"
          height="310px"
        />
        <div style={{ textAlign: 'center' }}>
          <span
            style={{
              color: 'yellow',
              paddingRight: '7.5px',
              textDecoration: 'none',
            }}
          >
            <Icon icon={faStar} />
            {item.highlighted_score.score}
          </span>
          <span
            style={{
              color: 'gray',
              paddingLeft: '7.5px',
              textDecoration: 'none',
            }}
          >
            <Icon icon={faUser} />
            {item.highlighted_score.formatted_amount_of_votes}
          </span>
        </div>
      </div>
    </Link>
  );
};

MovieListItem.propTypes = {
  elem: PropTypes.shape({
    onClick: PropTypes.func.isRequired,
    item: PropTypes.shape.isRequired,
  }),
};

export default MovieListItem;
