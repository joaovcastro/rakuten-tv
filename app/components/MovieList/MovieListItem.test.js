import React from 'react';
import { shallow } from 'enzyme';

import MovieListItem from './MovieListItem';

describe('<MovieListItem />', () => {
  const elem = {
    onClick: () => false,
    item: {
      id: 'item-id',
      images: {
        artwork: 'url',
      },
      highlighted_score: {
        score: 5,
      },
    },
  };

  it('correctly renders', () => {
    const wrapper = shallow(<MovieListItem elem={elem} />);
    expect(wrapper).toMatchSnapshot();
  });
});
