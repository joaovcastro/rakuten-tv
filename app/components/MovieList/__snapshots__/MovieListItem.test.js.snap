// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieListItem /> correctly renders 1`] = `
<Link
  to="/movie/item-id"
>
  <div
    className="grid"
    onClick={[Function]}
  >
    <img
      className="movie-poster"
      height="310px"
      src="url"
      width="225px"
    />
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <span
        style={
          Object {
            "color": "yellow",
            "paddingRight": "7.5px",
            "textDecoration": "none",
          }
        }
      >
        <Styled(FontAwesomeIcon)
          icon={
            Object {
              "icon": Array [
                576,
                512,
                Array [],
                "f005",
                "M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z",
              ],
              "iconName": "star",
              "prefix": "fas",
            }
          }
        />
        5
      </span>
      <span
        style={
          Object {
            "color": "gray",
            "paddingLeft": "7.5px",
            "textDecoration": "none",
          }
        }
      >
        <Styled(FontAwesomeIcon)
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f007",
                "M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z",
              ],
              "iconName": "user",
              "prefix": "fas",
            }
          }
        />
      </span>
    </div>
  </div>
</Link>
`;
